### DISCUSSION TITLE
Not able to get the cocept of Timestamp in lesson 9

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#620] [General]
i am not to understand what will be  the difference between block.timestamp and s_lasttimestamp when we assign
 ```
 s_lastTimeStamp=block.timestamp;
```
 and what does 
```
(block.timestamp-s_lastTimeStamp<i_interval
```
this statement mean then

**Comment by shikhar229169** [comment]
block.timestamp contains the value of current block time in unix format (in seconds).

```(block.timestamp - s_lastTimeStamp < i_interval``` This statement checks the diff of current block timestamp and the time s_lastTimeStamp was recently updated.

**Reply by shikhar229169** [reply]
Initially when you deploy the lottery  contract in constructor we are setting s_lastTimeStamp to the block time at which contract was deployed.
Suppose the timestamp at that moment was 1691907741, so s_lastTimeStamp = 1691907741

Now when you check (block.timestamp - s_lastTimeStamp < i_interval) after x seconds, after setting s_lastTimeStamp to 1691907741,
block.timestamp will give s_lastTimeStamp + x,
so it will be 1691907741 + x

**Reply by khushbusandhu** [reply]
got it
thanks @shikhar229169
